tests_e2e/tests/scripts/agent_firewall-verify_all_firewall_rules.py

```diff
@@ -0,0 +1,427 @@
+    except Exception as e:
+        raise Exception("Error -- failed to create HTTP connection with user:{0} \n {1}".format(user, e))
+
+    try:
+        blocked = False
+        client.request('GET', VERSIONS_PATH)
+    except Exception:
+        # if we get exception, it means the request is blocked
+        blocked = True
+
+    if not blocked:
+        raise Exception("Error -- unprivileged user:({0}) could connect to wireserver, make sure the firewall rules are set correctly".format(user))
+
+    log.info("Success -- unprivileged user:(%s) access to wireserver is blocked", user)
+
+
+def verify_http_to_wireserver_allowed(user: str) -> None:
+    """
+    This function is used to verify if http to wireserver is allowed for the given user
+    """
+    log.info("-----Verifying http request to wireserver is allowed")
+    switch_user(user)
+    try:
+        client = httpclient.HTTPConnection(get_wireserver_ip(), timeout=1)
+    except Exception as e:
+        raise Exception("Error -- failed to create HTTP connection with user:{0} \n {1}".format(user, e))
+
+    try:
+        client.request('GET', VERSIONS_PATH)
+    except Exception as e:
+        # if we get exception, it means the request is failed to connect
+        raise Exception("Error -- unprivileged user:({0}) access to wireserver failed:\n {1}".format(user, e))
+
+    log.info("Success -- privileged user:(%s) access to wireserver is allowed", user)
+
+
+def verify_non_root_accept_rule():
+    """
+    This function verifies the non root accept rule and make sure it is re added by agent after deletion
+    """
+    log.info("-----Verifying non root accept rule behavior")
+    log.info("Before deleting the non root accept rule , ensure a non root user can do a tcp to wireserver but cannot do a http request")
+    verify_dns_tcp_to_wireserver_is_allowed(NON_ROOT_USER)
+    verify_http_to_wireserver_blocked(NON_ROOT_USER)
+
+    # switch to root user required to stop the agent
+    switch_user(ROOT_USER)
+    # stop the agent, so that it won't re-add rules while checking
+    log.info("Stop Guest Agent service")
+    # agent-service is script name and stop is argument
+    stop_agent = ["agent-service", "stop"]
+    execute_cmd(stop_agent)
+
+    # deleting non root accept rule
+    non_root_accept_delete_cmd = get_non_root_accept_rule("-D")
+    delete_iptable_rules([non_root_accept_delete_cmd])
+    # verifying deletion successful
+    non_root_accept_check_cmd = get_non_root_accept_rule("-C")
+    verify_rules_deleted_successfully([non_root_accept_check_cmd])
+
+    log.info("** Current IP table rules\n")
+    print_current_iptable_rules()
+
+    log.info("After deleting the non root accept rule , ensure a non root user cannot do a tcp to wireserver request")
+    verify_dns_tcp_to_wireserver_is_blocked(NON_ROOT_USER)
+
+    switch_user(ROOT_USER)
+    # restart the agent to re-add the deleted rules
+    log.info("Restart Guest Agent service")
```

Comment:
I think we should add to this log why we're restarting the guest agent
```suggestion
    log.info("Restart Guest Agent service to re-add the deleted rules")
```